src/Snail/Lexer.hs

```diff
@@ -118,13 +118,14 @@ nonQuoteCharacter = do
 -- | ...
 textLiteral :: Parser SExpression
 textLiteral = do
+  sourcePosition <- getSourcePos
   text <- quotes (some $ escapedQuote <|> nonQuoteCharacter)
-  pure $ TextLiteral $ Text.concat text
+  pure $ TextLiteral (sourcePosition, Text.concat text)
 
 -- | ...
 sExpression :: Parser SExpression
 sExpression = SExpression <$> parens (lexemes `sepEndBy` spaces)
 
 -- | ...
 sExpressions :: Parser [SExpression]
-sExpressions = spaces *> sExpression `sepEndBy1` spaces
+sExpressions = (spaces *> sExpression `sepEndBy1` spaces) <* eof
```

Comment:
This is the bug fix. Basically, `()nil()` would parse `()` and fail on `nil` and `sExpressions` **wouldn't** consume the entire input. We should consume the entire input.